### PR TITLE
Pty improvements

### LINKED ISSUE #57: Improve dmsgpty
**Feature description**

Dmsgpty works well at the moment but it is a bit complicated to setup. We need to modify the bevaior in a few respects

- `dmsgpty-host` should default to production deployment `dmsg.discovery.skywire.skycoin.com` instead of `skywire.cc`
- `dmsgpty-host` should look for conf in `./config.json` by default and if no other path is specified
- `--skgen` should be deprecated. A new SecKey should be generated in all cases if a new conf is generated
- `--sk` should be deprecated. It should just be set via config file
- separate whitelisting files should be deprecated. The whitelisting values should be specified in config file
- `confpath` should be shortened to `-c`
- there is no need for an in memory whitelist. upon adding/removing whitelist items we should just flush to config

### PR DESCRIPTION
**Pull request by ersonp**
Fixes #57

 Changes:	
- moved the whitelist to config file code from dmsgpty-cli to dmsgpty-host
- added `NewConfigWhitelist()` to dmsgpty/whitelist.go for the host to read wl from the config
- added `configWhitelist` struct and the methods `All()`,  `Add()`, `Remove()` and `open()`
- removed viper dependency by merging the code from the PR #60 
- added instructions for dmsgpty in docs
- updated `github.com/creack/pty v1.1.9` to `github.com/creack/pty v1.1.10` because it was giving the following eror `Setctty set but Ctty not valid in child`

How to test this PR:
1. Run `make build`
2. Run `./bin/dmsgpty-host confgen`
3. Run `./bin/dmsgpty-host`
4. Run `./bin/dmsgpty-cli whitelist` or  `./bin/dmsgpty-cli whitelist-add public-key-goes-here` or `./bin/dmsgpty-cli whitelist-remove public-key-goes-here` on a different terminal
5. Run `./bin/dmsgpty-ui` and check `http://127.0.0.1:8080/`

### CODE REVIEW

**Review by jdknives — CHANGES_REQUESTED** [review]
I tried testing this PR and there are multiple issues. The config file is generated mostly fine, but we should also have a field for the PubKey of the host. There is no reason to only store the SecKey. 

Upon starting the dmsgpty host with a config file, the host just generates a new keypair instead of using the one from config. It should use the keypair from the config instead. It should not be possible to specify a config file as input that does not work. If it cant read or use the config, it should just fail.

The `--sk` flag should be remoed for security reasons.

File: docs/dmsgpty.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Dmsgpty
+`dmsgpty` is a utility exec which uses SSH tp connect to servers hosted over the `dmsg` network (similar to a simplified `SSH` over `dmsg`).
```
[the remark this answers is not in the document]
Comment (ersonp) [reply docs/dmsgpty.md]: The requested changes have been made. Please check and let me know if there are any more.

File: docs/dmsgpty.md [hunk 2]
```diff
@@ -57,10 +57,11 @@ $ ./bin/dmsgpty-cli --help
 
 ```
 
-### Example usage
-
+## Example usage
 In this example, we will use the `dmsg` network where the `dmsg.Discovery` address is `http://dmsg.discovery.skywire.skycoin.com`. However, any `dmsg.Discovery` would work.
 
+### Share Files
```
Comment (jdknives) [inline docs/dmsgpty.md]: Not sure why we would call this `Share Files`.

File: docs/dmsgpty.md [hunk 3]
```diff
@@ -107,4 +109,74 @@ To start the `dmsgpty-ui` simply run
 $ ./bin/dmsgpty-ui
 ```
 
-And open the browser at http://127.0.0.1:8080/
\ No newline at end of file
+And open the browser at http://127.0.0.1:8080/
+
+### Connect Two Local DMSG Hosts with DMSGPTY
```
Comment (jdknives) [inline docs/dmsgpty.md]: We need a documentation for connecting two remote machines with each other. This is a rare case so it should not be presented as the main case. It is sufficient to explain for this case how to modify the standard setup.

Comment (ersonp) [reply docs/dmsgpty.md]: Okay will change accordingly

File: docs/dmsgpty.md (hunk 2 above)
Comment (ersonp) [reply docs/dmsgpty.md]: Can you suggest a heading?

Comment (jdknives) [reply docs/dmsgpty.md]: We can just call it `Seting up remote command execution over dmsg`

Comment (ersonp) [reply docs/dmsgpty.md]: Okay

**Pull request merged**